packages/react/src/hooks/useAuth.ts

```diff
@@ -99,13 +99,13 @@ export const useAuth = (initialAuthStateOrOptions: UseAuthOptions = {}): UseAuth
   const initialAuthState = rest as any;
 
   const authContextFromHook = useAuthContext();
+  const isomorphicClerk = useIsomorphicClerkContext();
   let authContext = authContextFromHook;
 
-  if (authContext.sessionId === undefined && authContext.userId === undefined) {
+  if (!isomorphicClerk.loaded && authContext.sessionId === undefined && authContext.userId === undefined) {
     authContext = initialAuthState != null ? initialAuthState : {};
   }
```

Comment:
Should we flip this to 
```suggestion
  if (!isomorphicClerk.loaded && initialAuthState) {
    authContext = initialAuthState
  }
```

But also i'd expect that `authContextFromHook` has the correct data inside during SSR, since it get populated corrected in [here](https://github.com/clerk/javascript/blob/83eeb56de0d0e903dcb1d406b7668a6d11371f1e/packages/react/src/contexts/ClerkContextProvider.tsx#L40)